cardano-cli/src/Cardano/CLI/Mary/Parser.hs

```diff
@@ -0,0 +1,205 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Cardano.CLI.Mary.Parser
+  ( Token (..)
+  , Tokens
+  , TParser
+  , addition
+  , calculateValue
+  , lexToken
+  , lexTokens
+  , preValueAddition
+  , preValueLovelace
+  , preValueMultiAsset
+  , preValueParser
+  , preValtoValue
+  , stringToValue
+  , textToPolicyId
+  , valueTokenFullySpecified
+  , valueTokenPidAndAssetId
+  , valueTokenPidOnly
```

Comment:
This might be pedantic, but `pid` instantly makes me think of "process ID". 

Perhaps we can explicitly use `PolicyId` or something? Likely not a big deal, though.